### PR TITLE
feat: add date overrides to availability settings atom

### PR DESCRIPTION
**Pull request by Ryukemeister**
## What does this PR do?

 
- Adds date override feature for `Availability Settings` atom, this was previously not possible in v2 because of how the schedules api was structured. With the recent schedules refactor we now have date overrides for atoms as well.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

This can be tested in the platform examples app

### CODE REVIEW

File: packages/platform/atoms/availability/AvailabilitySettings.tsx [hunk 1]
```diff
@@ -507,23 +508,17 @@ export function AvailabilitySettings({
...
-                      ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"].indexOf(
-                        weekStart
-                      ) as 0 | 1 | 2 | 3 | 4 | 5 | 6
-                    }
-                  />
-                )}
-              </div>
-            ) : (
-              <></>
+            {enableOverrides && (
```
Comment (Ryukemeister) [inline packages/platform/atoms/availability/AvailabilitySettings.tsx:511]: `enableOverrides` prop allows users to enable or disable date overrides inside an atom

File: packages/platform/atoms/availability/atom-api-transformers/transformAtomScheduleForApi.ts [hunk 2]
```diff
@@ -13,9 +13,9 @@ export function transformAtomScheduleForApi(body: AvailabilityFormValues): Updat
     dateOverrides.flatMap(
       (dateOverridesRanges) =>
         dateOverridesRanges?.ranges?.map((range) => ({
-          date: `${range.start.getUTCFullYear}-${range.start.getUTCMonth}-${range.start.getUTCDate}`,
-          startTime: `${range.start.getUTCHours}-${range.start.getUTCMinutes}`,
-          endTime: `${range.end.getUTCHours}-${range.end.getUTCMinutes}`,
+          date: `${range.start.getUTCFullYear()}-${range.start.getUTCMonth()}-${range.start.getUTCDate()}`,
```
Comment (Ryukemeister) [inline packages/platform/atoms/availability/atom-api-transformers/transformAtomScheduleForApi.ts:16]: While transforming the overrides data into data suitable for v2 api none of the functions were being invoked and this was causing the requests to fail with an internal server error.

**Comment by supalarry** [comment]
There were problems when overrides were being formatted for API:
1. Date needed to have "+1" because Date.getMonth() returns values from 0 to 11, not 1 to 12.
2. The month, day of the date and start hours and end hours needed to be padded with zeroes.

In commit [fix: date overrides formatting for api request](https://github.com/calcom/cal.com/pull/15459/commits/58ad922714231e26fcc93ed0c279989f7213b2d7) I fixed it.

## Before
Screenshot 2024-06-18 at 09 59 46

## After
Screenshot 2024-06-18 at 10 15 38

**Review by supalarry — APPROVED** [review]
With fix above can confirm that it's possible to create date override and that it is reflected during booker's available times.

**Pull request merged**